### DISCUSSION TITLE
Problems upgrading to v2

### DISCUSSION THREAD

**Discussion by jensug** [mdx-js/mdx#2340] [Q&A] [answered]
Hello,

I have taken over a project  using mdx-jx and im trying to upgrade the dependencies but im having issues when it comes to mdx-js/loader because some of the methods used are outdated like the isMDXComponent function.

In my project MDX files have a meta prop whith a Layout component imported and other data like titles.

This is the file where problems occur (_app.js)

`import React, { useState, useEffect } from 'react'
import PropTypes from 'prop-types'
import { UIContext, defaultUIConfig } from '../components/contexts/ui'
import { useIsMd } from '../components/useIsScreenSize'
import '../styles/index.css'

const App = ({ Component, pageProps }) => {
    const isMd = useIsMd()
    const [uiConfig, setUIConfig] = useState({
        ...defaultUIConfig,
        showMenu: true,
        setShowMenu: value => setUIConfig(v => ({ ...v, showMenu: value })),
    })

    useEffect(() => {
        uiConfig.setShowMenu(!isMd)
    }, [isMd])

    if (Component.isMDXComponent) {
        const mdxProps = Component({}).props
        const meta = mdxProps.meta
        const Layout = meta.Layout

        return (
            <UIContext.Provider value={uiConfig}>
                <Layout meta={meta}>
                    <Component {...pageProps} />
                </Layout>
            </UIContext.Provider>
        )
    }

    return (
        <UIContext.Provider value={uiConfig}>
            <Component {...pageProps} />
        </UIContext.Provider>
    )
}`

And this is where the code creates some issues:

`    if (Component.isMDXComponent) {
        const mdxProps = Component({}).props
        const meta = mdxProps.meta
        const Layout = meta.Layout`

Any help is appreciated.
Thanks

**Comment by ChristianMurphy** [comment]
Welcome @jensug! 👋 
I'd recommend starting with the migration guide https://mdxjs.com/migrating/v2/
If the recommendations there don't meet your needs, take a bit of time to frame what you are actually trying to achieve by using these functions and props https://mdxjs.com/community/support/

**Comment by jensug** [comment]
Hi Christian,

Thanks for the reply and sorry for my lack of explaining. I have tried to follow the mirgration guide but failed to understood where i could have gone wrong, considering this project has fallen into my lap without the people who developed it being around anymore.

The props consist of a meta component with strings of titles and such. It also includes a component Layout which is wrapped around the mdx content, which has been working fine until the upgrade.

After upgrading to v2 the isMDXComponent function is obviously gone, so first thing would be how do i best change this? 
The other thing is component.props dont seem to consist of anything. Could there have been a change somewhere here?

Any help is appriciated.

**Reply by jensug** [reply]
Acctualy the props contaims the children but not the meta.

This is an example meta from an mdx

`
export const meta = {
    Layout,
    title: 'test',
    subtitle: 'test',
    courseId: 'test',
    topicId: 'test',
    lessonId: 'test',
}`

**Reply by ChristianMurphy** [reply]
@jensug taking a few steps back

1. what are you trying to achieve with the `meta` and `isMDXComponent`?
Are you trying to build a course/site structuring using the data? Is it used in the router? Is it used to edit the `<head>` tag similar to [react helmet](https://github.com/nfl/react-helmet)? or something else entirely?
2. how are you using the integration? You mention `mdx-js/loader` are you using that directly in a webpack file? Or using a Next/Gatsby wrapper?

---

My best guess reading between the lines of your comments, is that it is somewhere between:

> Are you trying to build a course/site structuring

and

> used to edit the `<head>` tag similar to [react helmet](https://github.com/nfl/react-helmet)

in which case, you just need to access the export, and the need for `isMDXComponent` is unclear (it's probably not needed).

Assuming you are leveraging webpack directly, make sure you are following the configuration guide at https://mdxjs.com/docs/getting-started/#webpack
Then you can import the `meta` directly.
```mdx
import Component, { meta } from './my-document.mdx'
```
and sent the `meta` data wherever you need it to go.

**Comment by jensug** [comment]
Hi, 

I am trying to build a course/site structuring.

The meta is just meta data for each course page mdx file containg titles and such. As well as the Layout component which is a setup i dont know why th eprevious developer chose. But i am not trying to add meta tags to the head.

I think the isMDXComponet was used for cases where pages were not supposed to be wrapped in a Layout.

Thanks!

**Reply by remcohaszing** [reply, marked as answer]
I just created [`recma-mdx-is-mdx-component`](https://github.com/remcohaszing/recma-mdx-is-mdx-component). You may want to use that for now.

**Comment by jensug** [comment]
I migth have found a solution where i put `
export default ({ children }) => <Layout meta={meta}>{children}</Layout>;  `at the end of every mdx file. It just involves copy pasting a lot of pages.

**Comment by jensug** [comment]
This post : https://github.com/mdx-js/mdx/issues/759 is looking at the same issue as i am.
